tests/stress/stress_test.go

```diff
@@ -122,3 +122,38 @@ func TestSync_SingleBlockProducer(t *testing.T) {
 		time.Sleep(time.Second)
 	}
 }
+
+func TestSync_SingleSyncingNode(t *testing.T) {
+	numNodes = 2
+	utils.SetLogLevel(log.LvlInfo)
+
+	// start block producing node
+	tmpdir, err := ioutil.TempDir("", "gossamer-stress-alice")
+	require.NoError(t, err)
+	defer os.Remove(tmpdir)
+	alice, err := utils.RunGossamer(t, 0, tmpdir, utils.GenesisDefault, utils.ConfigBABEMaxThreshold)
+	require.NoError(t, err)
+
+	time.Sleep(time.Second * 15)
+
+	// start syncing node
+	tmpdir, err = ioutil.TempDir("", "gossamer-stress-bob")
+	require.NoError(t, err)
+	defer os.Remove(tmpdir)
```

Comment:
you have os.Remove twice on defer for the tmpdir, perhaps you should then rename the vars so the defer will actually work properly. Otherwise it will defer remove the same dir twice ?

Reply:
if you run the test you can see that both directories get removed from `/tmp/`.

the `defer` captures the value of the args when it is reached and pushes the function call onto the stack, so the reassignment to `tmpdir` doesn't affect the first `os.Remove()`

Reply:
oh, I never tested this before. Good to know, Thanks :)